posthog/api/my_notifications.py

```diff
@@ -37,9 +37,10 @@ def get_unread(self, obj: ActivityLog) -> bool:
         if user_bookmark is None:
             return True
         else:
-            # API call from browser only includes milliseconds but python datetime in created_at includes microseconds
+           
+            # The frontend sends microseconds via toISOString() and we should respect that precision
             bookmark_date = user_bookmark.last_viewed_activity_date
-            return bookmark_date < obj.created_at.replace(microsecond=obj.created_at.microsecond // 1000 * 1000)
```

Comment:
we should add a test here that the case in the issue is covered by this fix

if it's not already covered then we should be covering

* something before the timestamp is considered read
* something at the timestamp exactly is considered read
* something after the timestamp is considered unread

i also wonder if we _should_ be truncating the timestamp
if an item was created at 223 microseconds after the bookmark date is it meaningful to say that it is unread?
activity log is not receiving writes at the kind of scale where that accuracy is helpful

maybe the fix is to truncate to the nearest second or half second on both sides of the comparison and have tests protecting that behaviour so the future traveller can see what is happening